palette/moreland/smooth.go

```diff
@@ -0,0 +1,279 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Some descriptive text is taken from http://www.kennethmoreland.com/,
+// ©2009--2016 Kenneth Moreland.
+
+// Color conversion functions are largely based on a spreadsheet
+// (http://www.kennethmoreland.com/color-maps/DivergingColorMapWorkshop.xls)
+// and a Python script (http://www.kennethmoreland.com/color-maps/diverging_map.py),
+// which are copyright their respective authors.
+
+// Package moreland provides color maps for pseudocoloring scalar fields.
+// The color maps are described at http://www.kennethmoreland.com/color-advice/
+// and in the following publications:
+//
+// "Why We Use Bad Color Maps and What You Can Do About It." Kenneth Moreland.
+// In Proceedings of Human Vision and Electronic Imaging (HVEI), 2016. (To appear)
+//
+// "Diverging Color Maps for Scientific Visualization." Kenneth Moreland.
+// In Proceedings of the 5th International Symposium on Visual Computing,
+// December 2009. DOI 10.1007/978-3-642-10520-3_9.
+package moreland
+
+import (
+	"fmt"
+	"image/color"
+	"math"
+
+	"github.com/gonum/plot/palette"
+)
+
+// smoothDiverging is a smooth diverging color palette as described in
+// "Diverging Color Maps for Scientific Visualization." by Kenneth Moreland,
+// in Proceedings of the 5th International Symposium on Visual Computing,
+// December 2009. DOI 10.1007/978-3-642-10520-3_9.
+type smoothDiverging struct {
```

Comment:
For the same reasons as above: it doesn't have any functionality that requires it to be exported and having it unexported decreases the API surface. But it can be exported if that's desired.